sources/Algo/src/Algo.cpp

```diff
@@ -81,16 +83,53 @@ GeneratorDefinitionAlgorithm::operator()(const NodePtr& node) {
   if (node_generators.size() == 1) {
     auto model = useInfiniteReactivelimits_ ? GeneratorDefinition::ModelType::SIGNALN : GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN;
     const auto& gen = node_generators.front();
+
+    if (model == GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN && !isDiagramValid(gen)) {
+      return;
+    }
+
     generators_.emplace_back(gen.id, model, node->id, gen.points, gen.qmin, gen.qmax, gen.pmin, gen.pmax);
   } else {
     auto model =
         useInfiniteReactivelimits_ ? GeneratorDefinition::ModelType::WITH_IMPEDANCE_SIGNALN : GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN;
     for (auto it = node_generators.begin(); it != node_generators.end(); ++it) {
+      if (model == GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN && !isDiagramValid(*it)) {
+        continue;
+      }
       generators_.emplace_back(it->id, model, node->id, it->points, it->qmin, it->qmax, it->pmin, it->pmax);
     }
   }
 }
 
+bool
+GeneratorDefinitionAlgorithm::isDiagramValid(const inputs::Generator& generator) {
+  bool valid = true;
+
+  if (generator.points.size() == 0) {
+    return true;
+  }
+  if (generator.points.size() == 1) {
+    LOG(warn) << MESS(InvalidDiagram, generator.id) << LOG_ENDL;
+    return false;
+  }
+
+  auto firstP = generator.points.front().p;
+  bool allQminEqualQmax = true;
+  bool allPEqual = true;
+  auto it = generator.points.begin();
+  while ((allQminEqualQmax || allPEqual) && it != generator.points.end()) {
+    allQminEqualQmax = allQminEqualQmax && it->qmin == it->qmax;
```

Comment:
as soon as this is false we might break the loop and avoir useless computations

Reply:
One Reactive Curve Point (RCP) which has qmin != qmax does not imply that the diagram is valid. This same diagram could have all its P equal for all we know, so we must continue the loop until we are sure.